adafruit_logging.py

```diff
@@ -101,63 +102,80 @@ def level_for(value: int) -> str:
     return LEVELS[0][1]
 
 
-class LoggingHandler:
+#  pylint: disable=too-few-public-methods
+class Handler:
     """Abstract logging message handler."""
 
-    def format(self, log_level: int, message: str) -> str:
+    def __init__(self, level=NOTSET):  # pylint: disable=undefined-variable
+        self.level = level
+        """Level of the handler; this is currently unused, and
+        only the level of the logger is used"""
+
+    def _format(self, log_level: int, message: str) -> str:
```

Comment:
are format and emit in this class ones that you mentioned changing to private because they don't exist in CPython?

I'm not super familiar with the logging module, but checking into CPython docs page it looks like it does have `emit()` and `format()` on the Handler class.
https://docs.python.org/3/library/logging.html#logging.Handler.emit

Reply:
It does have `emit()` and `format()`, but it's arguments are different. The way it is now, it takes the `log_level` parameter, but I could try to refactor it to take the same arguments.  I'd just have to figure out exactly what it getting passed as the `record` argument in CPython's `Handler.format()`.  Same goes for `Handler.emit()`.  My guess is that `record` is some kind of structure that captures a few things needed to log, including the level and message.

Reply:
Do you think that's worth it?  It might be nice to not do something to hacky like have methods with the same name but hidden, but I also am not sure what the use case is for directly calling a handler's `emit()` or `format()`.  Maybe for adding custom functionality?

Reply:
Okay, so the record that gets passed around is of type `LogRecord`.  It contains most of the info to handle converting into a message (minus things like time).  If we think it's worth it, we could add this class and change the relevant functions to use it.  The pro is that it means making custom handlers is possible!  The downside is more overhead and memory for what might otherwise typically be unused functionality.  Thoughts?

Reply:
In the long term I think it may be worthwhile to make our own LogRecord or atleast a subset of the CPython LogRecord so that we can have these functions with the same signature. 

But I'm also with you in that I don't know the use-case really for having emit and format functions like this. And I think it's okay to make this change for now and then follow up eventually with more changes, especially if we find folks that were or want to use this functionality.